### PR TITLE
chapter4n5

### PR DESCRIPTION
**Pull request by meimei0111**

### CODE REVIEW

**Comment by meimei0111** [comment]
ready.

**Comment by meimei0111** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+textarea {
+	resize: vertical;
+}
+
+.center {
+	text-align: center;
+	h1 {
+		margin-bottom: 10px;
+	}
+}
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: indenting only 2 spaces

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	font-size: 1.7em;
+	color: white;
+	text-transform: uppercase;
+	letter-spacing: -1px;
+	padding-top: 9px;
+	font-weight: bold;
+	&:hover {
+		color: white;
+		text-decoration: none;
+	}
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: indenting only 2 spaces

File: app/assets/stylesheets/custom.scss [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+  ul {
+    float: right;
+    list-style: none;
+    li {
+      float: left;
+      margin-left: 15px;
+     }
+  }
+}
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: image

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+class UsersController < ApplicationController
+  def new; end
+end
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]: image

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
-end
+  def full_title page_title
+    base_title = "Ruby on Rails Tutorial Sample App"
+    page_title.blank? ? base_title : "#{page_title} | #{base_title}"
+  end
+end
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]: image

File: app/views/layouts/_footer.html.erb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    <%= link_to t("ruby_on_rails_tut_author"), Settings.links.rails_tutorial.author %>
+  </small>
+  <nav>
+    <ul>
+      <li><%= link_to t(".about_footer"), about_path %></li>
+      <li><%= link_to t(".contact_footer"), "#" %></li>
+      <li><%= link_to t(".new_footer"), Settings.links.rails_tutorial.news %></li>
+    </ul>
+  </nav>
+</footer>
```
Comment (saobv-2223) [inline app/views/layouts/_footer.html.erb]: image

File: app/views/static_pages/home.html.erb [hunk 7]
```diff
@@ -1,8 +1,12 @@
...
-  <%= t ".description_detail" %>
-</p>
+  <p>
+  	<%= t ".description_title" %>
+  	<%= link_to t("ruby_on_rails_tut"), Settings.links.rails_tutorial.home_page %> 
+    <%= t ".description_detail" %>
+  </p>
+
+  <%= link_to t(".sign_up"), signup_path, class: "btn btn-lg btn-primary" %>
+</div>
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb]: image

File: app/views/users/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<% provide :title, t(".sign_up_title") %>
+<h1><%= t ".heading" %></h1>
+<p>
+  <%= t ".description_title" %>
+</p>
```
Comment (saobv-2223) [inline app/views/users/new.html.erb]: image

File: config/routes.rb [hunk 9]
```diff
@@ -2,8 +2,9 @@
...
-    get "static_pages/about"
-    get "static_pages/contact"
-  end
-end
+    get "/help", to: "static_pages#help"
+    get "/about", to: "static_pages#about"
+    get "/signup", to: "users#new"
+    get "users/new"
+  end 
+end
```
Comment (saobv-2223) [inline config/routes.rb]: image

**Comment by meimei0111** [comment]
ready.

**Comment by meimei0111** [comment]
ready.

**Comment by meimei0111** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
You must to fix all the comments then ready again

**Comment by meimei0111** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+textarea {
+      resize: vertical;
+}
+
+.center {
+      text-align: center;
+      h1 {
+              margin-bottom: 10px;
+      }
+}
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]:
```suggestion
body {
  padding-top: 60px;
}

section {
  overflow: auto;
}

textarea {
  resize: vertical;
}

.center {
  text-align: center;

  h1 {
    margin-bottom: 10px;
  }
}
```

File: app/assets/stylesheets/custom.scss [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+	color: white;
+	text-transform: uppercase;
+	letter-spacing: -1px;
+	padding-top: 9px;
+	font-weight: bold;
+	&:hover {
+		      color: white;
+		      text-decoration: none;
+	}
+}
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]:
```suggestion
#logo {
  float: left;
  margin-right: 10px;
  font-size: 1.7em;
  color: white;
  text-transform: uppercase;
  letter-spacing: -1px;
  padding-top: 9px;
  font-weight: bold;
  &:hover {
    color: white;
    text-decoration: none;
  }
}
```

**Comment by meimei0111** [comment]
ready

**Comment by meimei0111** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
merge

**Comment by meimei0111** [comment]
thank you!

**Pull request merged**